### PR TITLE
Feature/landing page map container

### PR DESCRIPTION
**Pull request by deye9**
The Landing Page will have a map of all the Locations. There should be a search bar, and clicking on a location should take you to its page. Location icons should be color coded: all online (green), some online (purple), all offline (red).
The map should be collapsible.

### CODE REVIEW

File: src/pages/home/locations.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// import { ResourceType } from '../../resource-type';
+// import { LOCATIONS_LIST_QUERY } from '../locations/queries';
+
+export const HomeLocations: React.FC = () => {
+  return (
+    <>
+      <LocationsList viewMode="map" height="320px" />
+    </>
+  );
+  // const [searchQuery, setSearchQuery] = useState('');
```
Comment (elliot-sabitov) [inline src/pages/home/locations.tsx:18]: can we revert these changes?

File: src/components/data-model-table/index.tsx [hunk 2]
```diff
@@ -19,6 +19,8 @@ export interface IDataModelListProps {
   hideCreateButton?: boolean;
   hideActions?: boolean;
   parentView?: ResourceType;
+  height?: string;
```
Comment (elliot-sabitov) [inline src/components/data-model-table/index.tsx:22]: I dont think we should need the height or the viewMode props. The height should be 100%, the map should grow and take up whatever size the parent container makes available. The viewMode I think for now is also not needed as a PROP because the LocationList for now will always render the table first... but I suppose it wouldnt hurt too much to leave this one in in cases where we want to show LocationList with the map as default... But the height, ideally, it would be helpful if it just automatically figured it out via css and was responsive to fill the parent container

File: src/pages/locations/index.tsx [hunk 3]
```diff
@@ -44,15 +46,50 @@ export const LocationsList: React.FC<IDataModelListProps> = (props) => {
...
   };
 
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filteredLocations, setFilteredLocations] = useState<Locations[]>([]);
+
+  const handleSearch = (query: string) => {
+    setSearchQuery(query);
+    const lowerCaseQuery = query.toLowerCase();
+
+    const filtered =
```
Comment (elliot-sabitov) [inline src/pages/locations/index.tsx:66]: Please Note: right now this may work because the dataSource will most likely give you all of the locations. In the future though, we will need to make this smarter so that it doesnt fetch all the locations and tries to display all of the locations on the map at once. Instead, the way it should work is that the map should identify the coordinates for the North East, North West, South East and South West, and send those values to the BE when fetching the locations, then we should only get and show the locations that are visible on the map.. when the user moves the map or zooms out, we would refetch. If we fetch all of the locations all at once, it can potentially overload the UI. For now though, I think we can just keep this in the back of our minds as something that we may need to optimize later.

File: src/pages/locations/index.tsx [hunk 4]
```diff
@@ -84,11 +132,67 @@ export const LocationsList: React.FC<IDataModelListProps> = (props) => {
...
+            justifyContent: 'center',
+            marginBottom: '60px',
+          }}
+        >
+          <div
+            style={{
+              position: 'relative',
+            }}
+          >
+            <input
```
Comment (elliot-sabitov) [inline src/pages/locations/index.tsx:150]: what do you think about using <AutoComplete /> here? - https://ant.design/components/auto-complete

File: src/pages/home/locations.tsx (hunk 1 above)
Comment (deye9) [reply src/pages/home/locations.tsx:18]: Done

File: src/pages/locations/index.tsx (hunk 3 above)
Comment (deye9) [reply src/pages/locations/index.tsx:66]: Technical debt noted

**Review by elliot-sabitov — APPROVED** [review]

**Review by thanaParis — APPROVED** [review]
Approved
